src/compiler.jl

```diff
@@ -2568,12 +2568,13 @@ function enzyme!(
             for u in LLVM.uses(fn)
                 ci = LLVM.user(u)
                 @assert isa(ci, LLVM.CallInst)
-                LLVM.replace_uses!(ci, operands(ci)[2])
-                LLVM.remove!(ci)
+                LLVM.replace_uses!(ci, operands(ci)[1])
+                LLVM.erase!(ci)
```

Comment:
You might need to collect the user instructions first since deleting the user may invalidate the iterator